designdocs/schemas_protocols_intents.md

```diff
@@ -0,0 +1,379 @@
+
+----------
+
+### 🏷 Intents and Protocols
+
+```rust
+pub type ProtocolInfos<T> = StorageMap<
+    _, Blake2_128Concat, ProtocolId, ProtocolInfo
+>;
+
+pub struct IntentInfo {
+    protocol_id: ProtocolId,
+    name: BoundedVec<u8, ConstU32<IDENTIFIER_MAX>>,
+    description: BoundedVec<u8, ConstU32<DESCRIPTION_MAX>>,
+    link: BoundedVec<u8, ConstU32<URL_MAX>>,
+    registered_at: BlockNumber,
+}
+
+pub type Intents<T> = StorageMap<
+    _, Blake2_128Concat, IntentId, IntentInfo
+>;
+
+// Q: Is this necessary as a separate map, or can be folded into the `IntentInfo` struct?
+pub type IntentVersions<T> = StorageMap<
+    _, Blake2_128Concat, IntentId, BoundedVec<SchemaId, MaxSchemaVersionsPerIntent>
+>;
+```
+
+Note: `IntentIds` are only referenced by numeric ID. Clients must resolve names via off-chain lookup using mappings
+from `ProtocolId/protocol name + intent name → IntentId`.
+
+----------
+
+### 🧹 Design Simplification
+
+The current runtime includes a top-level flag indicating whether any delegation exists between a user and provider. This
+was used to shortcut delegation checks. In this model, that summary flag is dropped — its only purpose was to optimize
+for the empty case. If need be, the design can be updated to preserve this structure.
+
+This simplification removes complexity and keeps delegation logic self-contained.
+
+## 10. **API Identifier Handling and Resolution** <a id="section_10"></a>
+
+The choice of identifiers in runtime and SDK APIs has significant implications for both usability and performance.
+Historically, Frequency APIs used `SchemaVersionId` for delegation and data interpretation. The proposed model must
+account for schemas and intents, while maintaining runtime efficiency.
+
+### 🔢 Preferred On-Chain Identifiers
+
+On-chain APIs and runtime logic should rely exclusively on numeric identifiers, ie `ProtocolId` and `IntentId` rather
+than `protocol_name.intent_namd`.
+
+String identifiers such as protocol names or intent names should never be parsed on-chain. These must
+be resolved by SDKs and off-chain services (e.g., the Frequency Gateway).
+
+### 🧠 Resolution Responsibility
+
+Clients and SDKs must perform resolution of strings to IDs before invoking extrinsics. For example:
+
+- Resolve `"dsnp.broadcast"` → `(ProtocolId, IntentId)`
+
+### 📬 Extrinsic Parameters
+
+Three possible design patterns:
+
+1. **Submit by SchemaId only (Legacy/Compatible):**
+
+    - Fast, direct lookup
+    - Requires on-chain schema metadata to determine intent for validation
+    - May require multi-delegation validation (intent or schema)
+
+2. **Submit by SchemaId + IntentId (Explicit):**
+
+    - Enables precision delegation checks
+    - Slightly more complex client integration
+
+3. **Submit by IntentId only (No Schema):**
+
+    - Enables non-schema-based actions (e.g., future custom actions)
+    - Runtime must validate delegation by `IntentId`
```

Comment:
from blockchain perspective, keeping things minimal, why we should support multiple ways to publish other than schemaId? that more granular for the chain and can enforce some provider side management or a library from us for schema management

Reply:
For clarification:

You _publish_ to a `SchemaId`; you _delegate_ to an `IntentId`. `Schemas` implement `Intents`. After thinking about it some more, I realized that any extrinsic that publishes to a Schema already has to read the `SchemaInfo` metadata to do some basic validation. So if the `IntentId` is part of that metadata, we don't even need an extra read to look it up to validate the delegation. Thus our existing extrinsic model of just passing `SchemaId` can remain.

Publishing to a schema on-chain, then, would look like:
* Read `SchemaInfo` by `SchemaId`
* Basic validation (ie, storage location)
* Get `IntentId` from `SchemaInfo`
* Query delegations for `IntentId` just as we currently would for `SchemaId`

Reply:
I'll update the doc to make ☝🏻 more clear